cpp/src_prims/score/scores.h

```diff
@@ -135,36 +134,56 @@ double trustworthiness_score(math_t *X, math_t *X_embedded, int n, int m, int d,
 
   int toDo = n;
   while (toDo > 0) {
-    int batchSize = min(toDo, MAX_BATCH_SIZE);
-    // Takes at most MAX_BATCH_SIZE vectors at a time
+    int curBatchSize = min(toDo, batchSize);
+
+    // Takes at most batchSize vectors at a time
+
+    size_t workspaceSize = 0;
 
     MLCommon::Distance::distance<distance_type, math_t, math_t, math_t,
                                  OutputTile_t>(
-      &X[(n - toDo) * m], X, d_pdist_tmp, batchSize, n, m, (void *)nullptr,
+      &X[(n - toDo) * m], X, d_pdist_tmp, curBatchSize, n, m, (void *)nullptr,
       workspaceSize, stream);
     CUDA_CHECK(cudaPeekAtLastError());
 
-    MLCommon::Selection::sortColumnsPerRow(d_pdist_tmp, d_ind_X_tmp, batchSize,
-                                           n, bAllocWorkspace, NULL,
-                                           workspaceSize, stream);
+    size_t colSortWorkspaceSize = 0;
+    bool bAllocWorkspace = false;
+    char *sortColsWorkspace;
+
+    MLCommon::Selection::sortColumnsPerRow(
+      d_pdist_tmp, d_ind_X_tmp, curBatchSize, n, bAllocWorkspace, nullptr,
+      colSortWorkspaceSize, stream);
+
+    if (bAllocWorkspace) {
```

Comment:
cuSolver and other libraries are pretty good at documenting the workspace needs, I think we should open an issue to improve our documentation to that standard, what do you think?

Reply:
That's a great idea! Created https://github.com/rapidsai/cuml/issues/1724